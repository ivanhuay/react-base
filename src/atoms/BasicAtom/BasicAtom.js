import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import * as styles from './BasicAtom.module.scss';

const cx = classNames.bind(styles);

const BasicAtom = ({ title }) => (
  <div className={cx('container')}><h2>{title}</h2></div>
);

BasicAtom.defaultProps = {
  title: '',
};

BasicAtom.propTypes = {
  title: PropTypes.string,
};
export default BasicAtom;
