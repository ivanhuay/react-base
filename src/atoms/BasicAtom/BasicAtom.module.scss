.container {
    padding: 20px;
    background: green;
}