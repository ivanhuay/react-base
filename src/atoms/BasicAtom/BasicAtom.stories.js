import React from 'react';

import BasicAtom from './BasicAtom';

export default {
  component: BasicAtom,
  title: 'atoms/BasicAtom',
};

export const Default = () => <BasicAtom>default</BasicAtom>;
