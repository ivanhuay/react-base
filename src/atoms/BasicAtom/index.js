import BasicAtom from './BasicAtom';

export default BasicAtom;
