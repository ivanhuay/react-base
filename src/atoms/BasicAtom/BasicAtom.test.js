import React from 'react';
import { render } from '@testing-library/react';
import BasicAtom from './BasicAtom';

test('Component render', () => {
  const { asFragment } = render(
    <BasicAtom title="example" />,
  );
  expect(asFragment()).toMatchSnapshot();
});
