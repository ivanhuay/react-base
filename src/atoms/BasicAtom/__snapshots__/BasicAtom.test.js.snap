// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      example
    </h2>
  </div>
</DocumentFragment>
`;
